project-koku/koku
# [COST-3546] Add option to return breakdown of distributed costs for a project

## Jira Ticket

[COST-3546](https://issues.redhat.com/browse/COST-3546)

## Description

This change will add the option to return a breakdown of the distributed costs for a project as `platform_distibuted` and `worker_unallocated_distributed` as well as the total.

## Testing

1. Checkout Branch
2. Restart Koku
3. load test data
4. Hit endpoint 
    ```
    http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/costs/?group_by[project]=*&group_by[cluster]=Test OCP on Premises
    ```

    1. You should see `platform_distributed`, `worker_unallocated_distributed` and `distributed` fields in the response similar to this;
    
    ```
        "cost": {
            "raw": {
                "value": 0.0,
                "units": "USD"
            },
            "markup": {
                "value": 0.0,
                "units": "USD"
            },
            "usage": {
                "value": 155.8624036716129,
                "units": "USD"
            },
            "platform_distributed": {
                "value": -155.8624036716129,
                "units": "USD"
            },
            "worker_unallocated_distributed": {
                "value": 0.0,
                "units": "USD"
            },
            "distributed": {
                "value": 0.0,
                "units": "USD"
            },
            "total": {
                "value": 155.8624036716129,
                "units": "USD"
            }
        }
    ```

## Notes

...
